cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs

```diff
@@ -1237,3 +1257,12 @@ pProtocolVersion =
             <> Opt.help "Minor protocol version. An increase indicates a soft fork\
                         \ (old software canvalidate but not produce new blocks)."
             )
+
+--------------------------------------------------------------------------------
+-- Helpers
+--------------------------------------------------------------------------------
+
+fromDouble :: Text -> Double -> UnitInterval
+fromDouble parser dbl = case mkUnitInterval $ toRational dbl of
+                   Just interval -> interval
+                   Nothing -> panic $ parser <> ": Please enter a value in the range [0,1]"
```

Comment:
We should be failing properly here, not calling `error` which means it has to use the `optparse-applicative` parser type for flag values `ReadM`

http://hackage.haskell.org/package/optparse-applicative-0.15.1.0/docs/Options-Applicative.html#t:ReadM

Note that `ReadM` is a `Monad` and is in `MonadFail` and other classes so you can do proper errors there.

So you should be able to say something like
```
pFieldUnitInterval :: ReadM UnitInterval
pFieldUnitInterval = Opt.auto >>= checkUnitInterval
  where
    checkUnitInterval :: Double -> ReadM UnitInterval
    checkUnitInterval dbl = case mkUnitInterval $ toRational dbl of
                   Just interval -> interval
                   Nothing -> fail "Please enter a value in the range [0,1]"
```